pkg/registry/thirdpartyresourcedata/codec.go

```diff
@@ -207,40 +208,49 @@ const template = `{
   "items": [ %s ]
 }`
 
-func encodeToJSON(obj *experimental.ThirdPartyResourceData) ([]byte, error) {
+func encodeToJSON(obj *experimental.ThirdPartyResourceData, stream io.Writer) error {
 	var objOut interface{}
 	if err := json.Unmarshal(obj.Data, &objOut); err != nil {
-		return nil, err
+		return err
 	}
 	objMap, ok := objOut.(map[string]interface{})
 	if !ok {
-		return nil, fmt.Errorf("unexpected type: %v", objOut)
+		return fmt.Errorf("unexpected type: %v", objOut)
 	}
 	objMap["metadata"] = obj.ObjectMeta
-	return json.Marshal(objMap)
+	encoder := json.NewEncoder(stream)
+	return encoder.Encode(objMap)
+}
+
+func (t *thirdPartyResourceDataCodec) Encode(obj runtime.Object) ([]byte, error) {
+	buff := &bytes.Buffer{}
```

Comment:
I'm pretty sure that's not true, since I can pass `&foo` into a function that expects `*foo` and then takes ownership of the value.  Any stack allocation has to be via pessimistic static analysis (e.g. no one ever asks for `&` of the variable, rather than anything else.